crates/biome_js_analyze/src/semantic_analyzers/nursery/no_empty_type_parameters.rs

```diff
@@ -0,0 +1,84 @@
+use crate::semantic_services::Semantic;
+use biome_analyze::{context::RuleContext, declare_rule, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_js_syntax::TextRange;
+use biome_js_syntax::{TsInterfaceDeclaration, TsTypeAliasDeclaration};
+use biome_rowan::declare_node_union;
+use biome_rowan::AstNode;
+use biome_rowan::AstSeparatedList;
+
+declare_rule! {
+    /// Disallow empty type parameters in type aliases and interfaces.
+    ///
+    /// TypeScript permits the use of empty type parameter lists in type alias and interface declarations; however, this practice is generally discouraged.
+    /// Allowing empty type parameter lists can lead to unclear or ambiguous code, where the intention of the generic type is not self-evident.
+    /// This rule disallows empty type parameter lists in type alias and interface declarations.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```ts,expect_diagnostic
+    /// interface Foo<> {}
+    /// ```
+    /// ```ts,expect_diagnostic
+    /// type Foo<> = {}
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```ts
+    /// interface Foo{}
+    /// ```
+    ///
+    /// ```ts
+    /// type Foo<T> = {
+    ///  bar: T;
+    /// }
+    /// ```
+    ///
+    pub(crate) NoEmptyTypeParameters {
+        version: "next",
+        name: "noEmptyTypeParameters",
+        recommended: true,
+    }
+}
+
+declare_node_union! {
+    pub(crate) NoEmptyTypeParametersQuery = TsInterfaceDeclaration | TsTypeAliasDeclaration
+}
+
+impl Rule for NoEmptyTypeParameters {
+    type Query = Semantic<NoEmptyTypeParametersQuery>;
```

Comment:
We don't need the semantic model, thus w could use `Ast` query.

```suggestion
    type Query = Ast<NoEmptyTypeParametersQuery>;
```

It is mostly a question of taste, however I could advise querying `TsTypeParameters` node instead of a custom union. You can then check if the parameter list's parent is an interface or a type alias:

```rs
matches!(node.syntax.parent().kinf(), Some(JsSyntaxKing::TS_INTERFACE_DECLARATION | JsSyntaxKing::TS_TYPE_ALIAS_DECLARATION))
```